scripts/src/main/resources/scripts/command/eclipse

```diff
@@ -124,72 +126,104 @@ function doCheckPluginsInstalled() {
       if [ -n "${new_plugin_dependencies}" ]
       then
         plugin_dependencies="${plugin_dependencies}\n${new_plugin_dependencies}"
       fi
       plugin=${plugin%.*}
     done
   done
-  # Remove plugin_dir from plugin variable
-  plugin_dependencies=$(echo -e "${plugin_dependencies}" | sed "s|${plugin_dir}/||g")
+  # Remove plugin_directory from plugin variable
+  plugin_dependencies=$(echo -e "${plugin_dependencies}" | sed "s|${plugin_directory}/||g")
   echo -e "${plugin_dependencies}" | sort -u
 }
 
 # Remove plugins from Eclipse by feature group ID
 function doRemovePlugin() {
-  local do_remove_plugin_id="${1}"
-  if [ -z "${do_remove_plugin_id}" ]
+  local rm_plugin_id="${1}"
+  if [ -z "${rm_plugin_id}" ]
   then
     doError "No plugin was specified for removal."
     return 1
   fi
-  doLoadPluginProperties "${do_remove_plugin_id}"
-  if [ -z "${plugin_id}" ]
+  doLoadPluginProperties "${rm_plugin_id}" # Sets global variables: plugin_id, plugin_url, plugin_name, plugin_status
+  if [ -z "${plugin_id}" ] 
   then
-    if [[ "${do_remove_plugin_id}" == *.* ]]
+    if [[ "${rm_plugin_id}" == *.* ]] # If rm_plugin_id contains a dot, it is probably a feature group ID
     then
-      plugin_id="${do_remove_plugin_id}"
+      plugin_id="${rm_plugin_id}"
     else
-      doError "Plugin ${do_remove_plugin_id} is not installed.\nMake sure you provided the correct plugin.properties name or plugin id."
+      doError "Plugin ${rm_plugin_id} is not installed.\nMake sure you provided the correct plugin.properties name or plugin id."
       return 1
     fi
   fi
   doEcho "Removing plugin ${plugin_id}..."
+  
+  local debugInfo
+  debugInfo=$(doInvokeEclipse org.eclipse.equinox.p2.director -uninstallIU "${plugin_id}")
+  
+  if [ "${debugInfo}" != "" ] && doCheckRemovalOfPluginDependencies "${debugInfo}" "${plugin_id}"
+  then
+    doSuccess "Plugin ${plugin_id} removed successfully."
+  fi
+}
+
+# Launches Eclipse with the given arguments and returns the debug info or empty String if the invocation failed.
+function doInvokeEclipse() {
+  local application="$1"
+  shift
+  local eclipse_args=("$@")
   local debug_info_file
   debug_info_file=$(mktemp)
-  "${ECLIPSE}" -nosplash -clean -consoleLog -vm "${ECLIPSE_JAVA_HOME}/bin/java" -application org.eclipse.equinox.p2.director -uninstallIU "${plugin_id}" -vmargs "${ECLIPSE_VMARGS}" >"${debug_info_file}" 2>&1
+  "${ECLIPSE}" -consoleLog -nosplash -vm "${ECLIPSE_JAVA_HOME}/bin/java" -application "${application}" "${eclipse_args[@]}" -vmargs "${ECLIPSE_VMARGS}" | grep '^!MESSAGE' | sed 's/^!MESSAGE //' > "${debug_info_file}" 2>&1
   local exit_code=$?
-  # Overwrite file with lines containing !MESSAGE
-  sed -i -n '/!MESSAGE/ p' "${debug_info_file}"
+  if [ $exit_code -ne 0 ];
+  then
+    doError "Failed to invoke Eclipse with arguments: ${eclipse_args[*]}"
+    echo ""
+    return 1
+  else
+  local debug_info
+  debug_info=$(<"${debug_info_file}")
```

Comment:
why do we first have to write the output into a file and then read the file into a variable and echo that?
IMHO this does not make sense anymore. Why don't we just remove `> "${debug_info_file}"` etc. and let the output go its regular way?

Reply:
Done